xinference/model/embedding/core.py

```diff
@@ -595,6 +601,49 @@ def encode(
 
             return all_embeddings, all_token_nums
 
+        @no_type_check
+        def _encode_clip(
+            model: SentenceTransformer,
+            sentences: Union[Dict[str, str], List[Dict[str, str]]],
+            convert_to_numpy: bool = True,
+            **kwargs,
+        ):
+            import base64
+            import re
+            from io import BytesIO
+
+            from PIL import Image
+
+            def base64_to_image(base64_str: str) -> Image.Image:
+                # base64_data = re.sub("^data:image/.+;base64,", "", base64_str)
+                base64_data = base64_str.split(",", 1)[1]
+                byte_data = base64.b64decode(base64_data)
+                image_data = BytesIO(byte_data)
+                img = Image.open(image_data)
+                return img
+
+            all_token_nums = 0
+            all_embeddings = []
+
+            for data in sentences:
+                if data.get("text") is not None:
+                    # texts.append(data['text'])
+                    txt_embed = model.encode([data["text"]])
+                    all_embeddings.append(txt_embed)
+                    all_token_nums += len(model.tokenize(data["text"]))
+                elif data.get("image") is not None:
+                    if re.match(r"^data:image/.+;base64,", data["image"]):
+                        image = base64_to_image(data["image"])
+                        # images.append(image)
+                        img_embed = model.encode([image])
+                        all_embeddings.append(img_embed)
+                    else:
+                        # images.append(data["image"])
+                        img_embed = model.encode([data["image"]])
+                        all_embeddings.append(img_embed)
```

Comment:
We used to have a know-how is that `model.encode(objects)` has better performance than doding `model.encode(obj)` one by one.

Is it possible to convert objects first, then do encode for one time to get the final embeddings?

Reply:
Correctly, I think there is the chance to batching encode ops.